### PR TITLE
chore(linter): update eslint-config-prettier

### LINKED ISSUE #4856: Update eslint-config-prettier in @nrwl/eslint-plugin-nx to Version 8.0.0
## Description
When updating a Project to eslint-config-prettier@8.0.0 all Linting Tasks Fail. 

This is because the @nrwl/eslint-plugin-nx package imports configs from eslint-config-prettier. 

In eslint-config-prettier@8.0.0 all configs were merged into one (see: [Changelog](https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md#version-800-2021-02-21)).

### PR DESCRIPTION
**Pull request by gperdomor**
## Related Issue(s)

Fixes #4856

### CODE REVIEW

**Review by elio-maggini — APPROVED** [review]
These are in fact the changes I made locally to my node_modules area. I was going to open a PR but you beat me to it :)

I can state that these changes fix the issue with prettier >= 8.0

File: packages/linter/src/migrations/update-10-3-0/update-eslint-builder-and-config.spec.ts
Comment (FrozenPandaz) [inline packages/linter/src/migrations/update-10-3-0/update-eslint-builder-and-config.spec.ts:33]: You don't need to change this.

File: packages/linter/src/migrations/update-10-4-0/update-eslint-configs-to-use-nx-presets.ts
Comment (FrozenPandaz) [inline packages/linter/src/migrations/update-10-4-0/update-eslint-configs-to-use-nx-presets.ts:396]: This might be a bit confusing but I don't think this should be removed.

We want this config here to include `prettier/@typescript-eslint` because it will be deduped out of the user's config. If it is removed, I believe users will still have `prettier/@typescript-eslint` in their config which will throw an error.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
Thank you for your contribution!

There's a few changes that need to be made.

Also, we need to migrate users to the new version of `eslint-config-prettier` as well by adding a package update here: https://github.com/nrwl/nx/blob/master/packages/linter/migrations.json#L40

**Comment by gperdomor** [comment]
> Also, we need to migrate users to the new version of `eslint-config-prettier` as well by adding a package update here: https://github.com/nrwl/nx/blob/master/packages/linter/migrations.json#L40

@FrozenPandaz Hi, can you help me with this please, I dont know if I should add the new entry inside packageJsonUpdates["10.3.0"].packages o create a new 10.4.0 entry?

**Comment by FrozenPandaz** [comment]
No worries, you'd add another property to the object.

It should look like this:
```json
"packageJsonUpdates": {
    "10.3.0": {
      ...
    },
    "11.5.0": {
      "version": "11.5.0-beta.0",
      "packages": {
        "eslint-config-prettier": {
          "version": "8.1.0",
          "alwaysAddToPackageJson": false
        }
      }
    }
  }
  ```
  
Also, in the case a user didn't run the migration and has an older version (below 8.0) of `eslint-config-prettier`, the new config removes some important rules...

My initial thought is we could add a `peerDependency` in the `package.json` so that users would be warned they are using an incompatible version. Any other ideas here?

**Comment by gperdomor** [comment]
@FrozenPandaz I updated the migrations like you suggested, in the case of the peer dependencies I think could be a good option, so that is included too

**Comment by gperdomor** [comment]
@FrozenPandaz can you trigger a new run for the failing test please?

**Comment by gperdomor** [comment]
Ping @FrozenPandaz 😊

**Comment by chrismcleod** [comment]
good work @gperdomor 

has anyone figured out a workaround until this is merged?  eslint in vscode doesn't seem to start at all and i've come to rely on react-rules-of-hook 😆

**Comment by elio-maggini** [comment]
@chrismcleod The "work around" is to roll back to a 7.X version of prettier in your package.json  :-D

**Comment by therk** [comment]
@FrozenPandaz what is pending to get this merged?

**Review by FrozenPandaz — APPROVED** [review]

**Comment by FrozenPandaz** [comment]
Thank you @gperdomor!

I made one small pedantic change to move the peer dependency to the eslint plugin but this will merge once tests pass :rocket: 

Thank you again!

**Comment by gperdomor** [comment]
@FrozenPandaz angular-e2e test fails 😢

**Comment by cyrus-za** [comment]
Thanks for this @gperdomor, @elio-maggini and @FrozenPandaz 
I was about to open a PR for this but then found this. I see this is in master. Which nx version will have it? I don't see it in the logs for the most recent releases.

**Comment by gperdomor** [comment]
@cyrus-za is included in Nx 11.5.1

**Pull request merged**
